core/src/main/java/org/frankframework/http/PushingListenerAdapter.java

```diff
@@ -109,13 +110,23 @@ public Message processRequest(Message rawMessage, PipeLineSession session) throw
 				log.debug("PushingListenerAdapter.processRequest() rethrows ListenerException...");
 				throw e;
 			}
+
 			log.debug("PushingListenerAdapter.processRequest() formats ListenerException to errormessage");
-			return handler.formatException(null, session, message, e);
+			return formatExceptionUsingErrorMessageFormatter(session, message, e);
 		} finally {
 			ThreadContext.clearAll();
 		}
 	}
 
+	// The ApplicationContext is practically always an Adapter except when the listener is created directly via the LarvaScenarioContext
+	private Message formatExceptionUsingErrorMessageFormatter(PipeLineSession session, Message inputMessage, Throwable t) {
+		if (applicationContext instanceof Adapter adapter) {
+			return adapter.formatErrorMessage(null, t, inputMessage, session, null);
+		}
+
+		log.warn("unformatted exception while processing input request [{}]", inputMessage, t);
+		return new Message(t.getMessage());
+	}
```

Comment:
Although over here we can safely assume the `ApplicationContext` is always an `Adapter`, I've copied this block from the `JavaListener`.